test/js/circuit_breaker.ts

```diff
@@ -176,5 +176,36 @@ describe("circuit", function() {
     await sleep(1500);
     assert.equal(loggerCalls, 1);
   });
-});
 
+  it("applies callback, if provided, to circuit-breaker error", async () => {
+    const c = new Client({
+      address: mockAddress,
+      circuit: {
+        forceClosed: false,
+        requestVolumeThreshold: 5,
+      },
+      logger: {
+        errorD: () => {},
+        infoD: () => {},
+      },
+      retryPolicy: RetryPolicies.None,
+    });
+
+    for (let i = 0; i < 10; i++) {
+      const mockApi = nock(mockAddress).get("/v1/authors").replyWithError("Yikes");
+      // The `await` here is necessary to surface assertion errors in the callback. Without the
+      // `await`, assertion errors will generate unhandled promise rejections and be swallowed
```

Comment:
Instead of mixing callbacks and promises here in a way that is a little confusing I think even with the comment, why not change line 180 to have a `done` callback and call it in the callback once we've asserted.

Reply:
Good point! I tried using `done` earlier but ran into some snags. Lemme give it another shot though with a fresh mind